src/compiler/factoryPublic.ts

```diff
@@ -1412,9 +1412,11 @@ namespace ts {
     }
 
     export function updateBinary(node: BinaryExpression, left: Expression, right: Expression, operator?: BinaryOperator | BinaryOperatorToken) {
+        const nextOperator = operator ?? node.operatorToken;
         return node.left !== left
             || node.right !== right
-            ? updateNode(createBinary(left, operator || node.operatorToken, right), node)
+            || node.operatorToken !== nextOperator

```

Comment:
If you take the suggestion:

```suggestion
            || node.operatorToken !== operator
```